Products/CMFPlone/browser/exceptions.py

```diff
@@ -31,13 +34,17 @@ def __call__(self):
                 'error_type': error_type,
             })
 
-        # Use a simplified template if main_template is not available
-        try:
-            self.context.unrestrictedTraverse('main_template')
-        except:
+        if IApplication.providedBy(self.context):
```

Comment:
Rendering the template is going to be a problem not just on the Application, but also anything outside of a Plone site. Instead of this check I would suggest checking whether you can get the portal.